packages/app-storage/src/Selection/Modules.tsx

```diff
@@ -80,21 +82,58 @@ class Modules extends TxComponent<Props, State> {
   }
 
   private nextState (newState: Partial<State>): void {
+    const { api } = this.props;
+
     this.setState(
       (prevState: State): Pick<State, never> => {
         const { key = prevState.key, values = prevState.values } = newState;
-        const hasParam = key.creator.meta.type.isMap;
-        const isValid = values.length === (hasParam ? 1 : 0) &&
-          values.reduce(
-            (isValid, value): boolean =>
+
+        const areParamsValid = (): boolean => {
+          return values.reduce(
+            (isValid: boolean, value): boolean => (
               isValid &&
               !isUndefined(value) &&
               !isUndefined(value.value) &&
-              value.isValid,
+              value.isValid),
             true
           );
+        };
+
+        if (key.creator.meta.type.isDoubleMap) {
+          let defaultValues = null;
+          const key1 = key.creator.meta.type.asDoubleMap.key1.toString();
+          const key2 = key.creator.meta.type.asDoubleMap.key2.toString();
+
+          if (key !== prevState.key && key.creator.section === 'session' && key.creator.method === 'nextKeys') {
```

Comment:
Probably would make sense to split this into a seperate func, sure there will be more. (And the checks here will grow)